### PR TITLE
Type change int -> long to prevent tranche novel variant count overflow

### PR DESCRIPTION
**Pull request by ldgauthier**
in VQSR scattered mode

### CODE REVIEW

**Comment by mcovarr** [comment]
Would it be difficult to write a test for this with `numKnown` and `numNovel` > `Integer.MAX_VALUE` just to make sure all the required spots have been updated?

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/Tranche.java [hunk 1]
```diff
@@ -27,14 +27,16 @@ public class Tranche {
...
-    final int numNovel;      //number of novel sites in this tranche
+    final long numKnown;      //number of known sites in this tranche
+    final long numNovel;      //number of novel sites in this tranche
     final VariantRecalibratorArgumentCollection.Mode model;
     final String name;       //Name of the tranche
 
-    public Tranche(final String name, final double knownTiTv, final int numNovel, final double minVQSLod, final VariantRecalibratorArgumentCollection.Mode model, final double novelTiTv, final int accessibleTruthSites, final int numKnown, final int callsAtTruthSites) {
+    public Tranche(final String name, final double knownTiTv, final long numNovel, final double minVQSLod,
+                   final VariantRecalibratorArgumentCollection.Mode model, final double novelTiTv,
+                   final int accessibleTruthSites, final long numKnown, final int callsAtTruthSites) {
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/Tranche.java:37]: Did you check the upstream call sites for uses of `int` to store the number of known/novel sites? If there's an `int` anywhere in the call chain the overflow issues will likely persist...

**Review by droazen — CHANGES_REQUESTED** [review]
One question for you @ldgauthier

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/Tranche.java (hunk 1 above)
Comment (ldgauthier) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/Tranche.java:37]: I found one in the input parsing and fixed it.  (The number of truth sites values are also ints, but with the resources we use today they're way, way below Integer.MAX_VALUE and I don't expect that to ever change based on the size of the human genome.)

**Comment by ldgauthier** [comment]
@mcovarr added two tests (and refactored a dataprovider, because I'm a good person) -- take another look?

**Review by mcovarr — APPROVED** [review]
LGTM! 👍

**Comment by ldgauthier** [comment]
Closes #7859

**Review by droazen — APPROVED** [review]
👍 Looks good @ldgauthier !

**Pull request merged**